### PR TITLE
feat: context menu for stack sections

### PR DESCRIPTION
**Pull request by jessmartin**
https://github.com/dxos/dxos/assets/27258/50a02d36-a90c-45e4-8d0a-1996e7c6c3a9

## Motivation

Stack items need to have other options than just drag and remove from stack. The next option will be "Navigate to item." 

In order to prepare for arbitrary context menu actions on a stack section, replace the "remove section" button with a context menu and move the "remove section" option into the context menu.

## Finish Line

- [x] replace remove button with a context menu and move remove section into the context menu
- [x] context menu (triple dot) should only show on hover
- [x] add "go to item" as menu item that navigates to the item
- [ ] ~~make it possible for plugins to add arbitrary actions to the stack section~~

### Summary
🎨♻️🗑️

Refactored section menu UI and logic in `Section.tsx` to use existing components and simplify code.

> _`SectionMenu` is gone, replaced by `Card.Menu`_
> _No more clutter, no more confusion, only `DropdownMenu.Item`_
> _We purge the code of props and styles that we don't need_
> _We make it consistent, we make it accessible, we make it bleed_

### Walkthrough
* Replace `Button` and `X` components with `Card.Menu` and `DropdownMenu.Item` for section menu in `Section.tsx` ([link](https://github.com/dxos/dxos/pull/4722/files?diff=unified&w=0#diff-ee852658cb7f1e4f27e313e116c6246b994ece1114a8da475447b9c9beee5243L8-R8))
* Move `onClick` handler for removing section to `DropdownMenu.Item` and remove unnecessary props from `Button` in `Section.tsx` ([link](https://github.com/dxos/dxos/pull/4722/files?diff=unified&w=0#diff-ee852658cb7f1e4f27e313e116c6246b994ece1114a8da475447b9c9beee5243L80-R94))
* Update `classNames` prop to use `fineButtonDimensions` utility and adjust styles for menu button and active state in `Section.tsx` ([link](https://github.com/dxos/dxos/pull/4722/files?diff=unified&w=0#diff-ee852658cb7f1e4f27e313e116c6246b994ece1114a8da475447b9c9beee5243L80-R94))
* Set `position` prop to `undefined` for `Card.Menu` to use default position in `Section.tsx` ([link](https://github.com/dxos/dxos/pull/4722/files?diff=unified&w=0#diff-ee852658cb7f1e4f27e313e116c6246b994ece1114a8da475447b9c9beee5243L80-R94))

### CODE REVIEW

File: packages/ui/react-ui-stack/src/components/Section.tsx [hunk 1]
```diff
@@ -77,18 +77,21 @@ export const Section = forwardRef<HTMLLIElement, SectionProps>(
...
-            <Button
-              variant='ghost'
-              classNames={[
-                'self-stretch justify-start rounded-is-none',
+            <Card.Menu
+              position={undefined}
+              classNames={mx(
+                fineButtonDimensions,
+                'self-stretch justify-center rounded-is flex items-center bs-auto is-auto',
                 hoverableFocusedControls,
```
Comment (jessmartin) [inline packages/ui/react-ui-stack/src/components/Section.tsx:85]: For some reason the hover behavior changed such that the menu shows all of the time.

Comment (jessmartin) [reply packages/ui/react-ui-stack/src/components/Section.tsx]: fixed it

**Comment by jessmartin** [comment]
As for `make it possible for plugins to add arbitrary actions to the stack section`, I think that can hold off for another PR. I don't have any arbitrary actions I want to add just the second.

File: packages/ui/react-ui-mosaic/src/mosaic/Tile.tsx [hunk 2]
```diff
@@ -43,6 +43,7 @@ export type MosaicTileProps<TData extends MosaicDataItem = MosaicDataItem, TPosi
     // TODO(burdon): Generalize events via onAction?
     onSelect?: () => void;
     onRemove?: () => void;
+    onNavigate?: () => void;
```
Comment (jessmartin) [inline packages/ui/react-ui-mosaic/src/mosaic/Tile.tsx:46]: Should I be using `onAction` instead? I couldn't find many other examples.

Comment (thure) [reply packages/ui/react-ui-mosaic/src/mosaic/Tile.tsx:46]: That is a to-do that we can resolve another time, IMO.

File: packages/ui/react-ui-stack/src/components/Section.tsx [hunk 3]
```diff
@@ -76,19 +79,48 @@ export const Section = forwardRef<HTMLLIElement, SectionProps>(
...
+                    active === 'overlay' && 'invisible',
+                  ]}
+                  // data-testid={testId}
+                >
+                  <DotsThreeVertical className={getSize(4)} />
+                </Button>
+              </DropdownMenu.Trigger>
+
+              <DropdownMenu.Portal>
+                <DropdownMenu.Content classNames='z-[31]'>
```
Comment (thure) [inline packages/ui/react-ui-stack/src/components/Section.tsx:110]: You shouldn't need to set z-index if your content is within a `Portal` (I realize even my own code does this) — this isn't blocking, but could you try removing the z-index and see if it still works as expected?

**Review by thure — APPROVED** [review]
lgtm!

File: packages/ui/react-ui-stack/src/components/Section.tsx (hunk 3 above)
Comment (jessmartin) [reply packages/ui/react-ui-stack/src/components/Section.tsx:110]: sure! will give it a try.

Comment (jessmartin) [reply packages/ui/react-ui-stack/src/components/Section.tsx]: works great

**Pull request merged**